File: desktop/src/ui/strings.ts
// The 14 strings that make up the entire pairing-flow vocabulary.
// Any user-visible text not in this file is a bug — see PAIRING.md.

export const STRINGS = {
  // PC home, before phone is seen
  "home.pc.idle": "Open Tether on your phone.",

  // Phone home, before PC is seen (lives here too so we can mirror
  // the table on the Android side without drift)
  "home.phone.idle": "Looking for your computer…",

  // Cascade status lines
  "cascade.mdns": "Looking on your Wi-Fi…",
  "cascade.fallback": "Trying another way…",
  "cascade.usb.prompt": "Connect your phone with any USB cable.",
  "cascade.usb.detected": "Got it. Finishing up…",
  "cascade.usb.debug": "Tap below to turn on USB debugging — it takes 10 seconds.",
  "cascade.hotspot": "Turn on your phone's hotspot. We'll connect to it automatically.",

  // Pairing card
  "pair.card.title": "Pair with {peer}?",
  "pair.card.subhead":
    "These three emojis should match on both screens. Tap Confirm on both.",
  "pair.card.confirm": "Confirm pairing",
  "pair.mismatch":
    "These emojis don't match. Don't confirm — start over from both apps.",
  "pair.success": "You're paired. You can close this.",
  "pair.manual": "Pair another way",
} as const;

export type StringKey = keyof typeof STRINGS;

export function t(key: StringKey, vars?: Record<string, string>): string {
  let s = STRINGS[key] as string;
  if (vars) {
    for (const [k, v] of Object.entries(vars)) {
      s = s.replace(`{${k}}`, v);
    }
  }
  return s;
}

File: android/app/src/main/kotlin/io/tether/ui/Strings.kt
package io.tether.ui

// Mirror of desktop/src/ui/strings.ts. Any user-visible text not in
// this map is a bug — see PAIRING.md.

object Strings {
    const val HomePcIdle = "Open Tether on your phone."
    const val HomePhoneIdle = "Looking for your computer…"

    const val CascadeMdns = "Looking on your Wi-Fi…"
    const val CascadeFallback = "Trying another way…"
    const val CascadeUsbPrompt = "Connect your phone with any USB cable."
    const val CascadeUsbDetected = "Got it. Finishing up…"
    const val CascadeUsbDebug =
        "Tap below to turn on USB debugging — it takes 10 seconds."
    const val CascadeHotspot =
        "Turn on your phone's hotspot. We'll connect to it automatically."

    const val PairCardSubhead =
        "These three emojis should match on both screens. Tap Confirm on both."
    const val PairCardConfirm = "Confirm pairing"
    const val PairMismatch =
        "These emojis don't match. Don't confirm — start over from both apps."
    const val PairSuccess = "You're paired. You can close this."
    const val PairManual = "Pair another way"

    fun pairCardTitle(peer: String) = "Pair with $peer?"
}
